src/Symfony/Component/Yaml/Parser.php

```diff
@@ -32,18 +32,19 @@ class Parser
     private $skippedLineNumbers = array();
     private $locallySkippedLineNumbers = array();
 
-    /**
-     * Constructor.
-     *
-     * @param int      $offset             The offset of YAML document (used for line numbers in error messages)
-     * @param int|null $totalNumberOfLines The overall number of lines being parsed
-     * @param int[]    $skippedLineNumbers Number of comment lines that have been skipped by the parser
-     */
-    public function __construct($offset = 0, $totalNumberOfLines = null, array $skippedLineNumbers = array())
+    public function __construct()
     {
-        $this->offset = $offset;
-        $this->totalNumberOfLines = $totalNumberOfLines;
-        $this->skippedLineNumbers = $skippedLineNumbers;
+        if (func_num_args() > 0) {
```

Comment:
instead of having to do a `if func > 0`, `if func_args > 1`, ..., you can leave the optionnal parameters in the constructor (just undocument those) and just check if `func_num_args > 0` to trigger the deprecation :

```php
<?php
$f = function ($a = null, $b = null, array $c = array())
{
    var_dump(func_num_args());

    if (func_num_args() > 0) {
      trigger();
    }

   $this->a = $a;
   // ...
}

$f(); // output 0, no deprecation
$f(1); // output 1, deprecation
// and so on
```

It's cleaner IMO (and less conditionnals too)